feat: documenting tokens list v2, search v2, detokenize and stripe forward endpoints (#415)
Basis-Theory/developers.basistheory.com: docs/api/tokens/detokenize.mdx

---
title: Detokenize
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

import {
  HttpMethod,
  HttpMethods,
} from "@site/src/components/shared/HttpMethod";
import { Permission } from "@site/src/components/shared/Permission";

# Detokenize

The `detokenize` endpoint enables you to [detokenize](/docs/expressions/detokenization) tokens in order to retrieve their original values.

The maximum number of tokens that can be detokenized within a single request is defined by your Tenant's "Detokenize Limit" [Quota](/docs/api/rate-limits#default-quotas).

<HttpMethod
  method={HttpMethods.POST}
  endpoint="https://api.basistheory.com/detokenize"
/>

#### Permissions

<Permission content="token:read" />

<br/><br/>

> The `token:read` permission must be granted on each token container, and the required scope of this permission
> is determined by the [containers](/docs/concepts/what-are-containers) of those tokens being detokenized in the request.

Reviewer: I'm not sure how to reword it, but `the required scope of this permission is determined by" doesn't make sense to me - maybe you don't need that second part of the sentence at all

```suggestion
The `token:read` permission must be granted on the [container](/docs/concepts/what-are-containers) of each token included in the request.
```

    "ApiKey": {Key: "<API_KEY>"},
  })

    tokens, httpResponse, err := apiClient.TokenizeApi.Tokenize(contextWithAPIKey).Body(map[string]interface{}{
    "tokens": [2]string{"7993dbd8-19d5-44bf-b6dc-078e8b44c6cb", "a1883f97-2e4c-439a-bf73-dffb681de929"}
  }).Execute()
  }
    ```

  </TabItem>
</Tabs>

### Response

```json showLineNumbers
{
  "tokens": [
      {
  "id": "7993dbd8-19d5-44bf-b6dc-078e8b44c6cb",
  "type": "token",
  "tenant_id": "77cb0024-123e-41a8-8ff8-a3d5a0fa8a08",
  "data": "secret data",
  "containers": ["/general/high/"],
  "metadata": {
    "nonSensitiveField": "Non-Sensitive Value"
  },
  "created_by": "fb124bba-f90d-45f0-9a59-5edca27b3b4a",
  "created_at": "2021-03-01T08:23:14+00:00"
},
{
  "id": "a1883f97-2e4c-439a-bf73-dffb681de929",
  "type": "token",
  "tenant_id": "77cb0024-123e-41a8-8ff8-a3d5a0fa8a08",
  "data": "secret data 2",
  "containers": ["/general/high/"],
  "metadata": {
    "nonSensitiveField": "Non-Sensitive Value 2"
  },
  "created_by": "fb124bba-f90d-45f0-9a59-5edca27b3b4a",
  "created_at": "2021-03-01T08:23:14+00:00"
}
  ]

> }

Reviewer 2: Can you fix the indentation on this example?
